### PR TITLE
Implemented Auxiliary Chain model and related test cases.

### LINKED ISSUE #54: Implement auxiliary chains model class
Implement `auxiliary_chains` model class

- chain_id (PK)
- origin_chain_name
- last_processed_block_number
- last_origin_block_height
- (last_origin_state_root)
- last_auxiliary_block_height
- (last_auxiliary_state_root)
- ost_gateway_address (OSTPrime gateway)
- ost_cogateway_address (OSTPrime CoGateway)
- anchor_address
- coanchor_address

Abstract the database interaction layer.

### PR DESCRIPTION
**Pull request by puneet-khushwani-eth**
Fixes #54

### CODE REVIEW

File: tests/models/AuxiliaryChainRepository/create.test.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  });
+
+  it('Checks creation of auxiliary chain model.', async (): Promise<void> => {
+    const auxiliaryChainAttributes: AuxiliaryChainAttributes = {
+      chainId: 10001,
+      originChainName: '10001',
+      ostGatewayAddress: '0x497A49648885f7aaC3d761817F191ee1AFAF399C',
+      ostCoGatewayAddress: '0x497A49648885f7aaB3d761817F191ee1AFAF399C',
+      anchorAddress: '0x3F45616cFb992988943ff3bA00e8c0aA46B4540a',
+      coAnchorAddress: '0xD1008015aA0Cf2a61493Bd19dE6C9ca88E934FCe',
```
Comment (deepesh-kn) [inline tests/models/AuxiliaryChainRepository/create.test.ts]: This address looks like actual addresses, for testing, we should use it like
`0x000000000000000000000000000000000000001`

File: tests/models/AuxiliaryChainRepository/util.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+const Util = {
+  checkAuxiliaryChainAgainstAttributes(
+    auxiliaryChain: AuxiliaryChain,
+    auxiliaryChainAttributes: AuxiliaryChainAttributes,
+  ): void {
+    assert.strictEqual(
+      auxiliaryChain.chainId,
+      auxiliaryChainAttributes.chainId,
+      // 'chainId should match',
```
Comment (deepesh-kn) [inline tests/models/AuxiliaryChainRepository/util.ts]: Any reason to comment this out?

File: src/models/AuxiliaryChainRepository.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        coAnchorAddress: {
+          type: DataTypes.STRING,
+          allowNull: false,
+          validate: {
+            isAlphanumeric: true,
+            len: [42, 42],
+          },
+        },
+        lastProcessedBlockNumber: {
+          type: DataTypes.INTEGER,
```
Comment (deepesh-kn) [inline src/models/AuxiliaryChainRepository.ts]: What is the max for `DataTypes.INTEGER`?
Do we have BIGINT type in DB?

Comment (puneet-khushwani-eth) [reply src/models/AuxiliaryChainRepository.ts]: 2,147,483,647 is the max for DataTypes.INTEGER. 

Agree that we should change the BlockNumber related variables to BIGINT.

File: tests/models/AuxiliaryChainRepository/util.ts (hunk 2 above)
Comment (puneet-khushwani-eth) [reply tests/models/AuxiliaryChainRepository/util.ts]: This was a typo. Fixed it.

File: src/models/AuxiliaryChainRepository.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+ */
+export interface AuxiliaryChainAttributes {
+  chainId: number;
+  originChainName: string;
+  ostGatewayAddress: string;
+  ostCoGatewayAddress: string;
+  anchorAddress: string;
+  coAnchorAddress: string;
+  lastProcessedBlockNumber?: number;
+  lastOriginBlockHeight?: number;
```
Comment (deepesh-kn) [inline src/models/AuxiliaryChainRepository.ts]: The types should be BigNumber too. `BN`.
Can you please check all the attributes of already merged models.

**Comment by puneet-khushwani-eth** [comment]
Looks like Travis is facing some problems downloading nvm dependencies. Have verified that these test cases run successfully on local machine.

**Pull request merged**